### PR TITLE
Only install directories if tftp is managed

### PR DESCRIPTION
**Pull request by dc-mattj**
Only install the directories and pxe files if tftp_managed is true.

### CODE REVIEW

**Comment by mmoll** [comment]
This is a bit against what I wanted with introducing the paramter, however seeing this now, I think it would probably be better to move some things from here into our tftp module, but would like to hear other maintainer's voices.

The backgroup back then was that I want to use foreman-installer to install everything needed to have a libvirt compute resource with DNS/DHCP enabled and libvirt used as TFTP server (so the regular TFTP daemon should not get installed), as that's a common small setup. When testing this on a fresh, clean system last week, however, I already ran into the problem that syslinux wasn't installed and one (or more?) needed directories were not created.

**Comment by domcleal** [comment]
I think the idea of the change is fine, but it should probably move to `foreman_proxy::config` in the same way that `dhcp_managed` works.

**Comment by mmoll** [comment]
@dc-mattj what's the status here?

**Comment by dc-mattj** [comment]
Sorry, lost track of this one slightly. Will take a look tomorrow at the foreman_proxy::config stuff and refactor the change.

**Comment by mmoll** [comment]
lint fails with:
```
manifests/tftp.pp:18:WARNING: top-scope variable being used without an explicit namespace
manifests/tftp.pp:3:ERROR: trailing whitespace found
```

**Comment by dc-mattj** [comment]
Sorry, my bad - fixed now

File: manifests/tftp.pp [hunk 1]
```diff
@@ -1,32 +1,25 @@
...
-  $tftp_managed = $foreman_proxy::tftp_managed,
-) {
-  $required_tftp_classes = $tftp_managed ? {
-    true    => Class['foreman_proxy::install', 'tftp::install'],
-    default => Class['foreman_proxy::install'],
+class foreman_proxy::tftp {
+
+  class { '::tftp':
+    root => $foreman_proxy::tftp_root,
   }
```
Comment (mmoll) [inline manifests/tftp.pp]: superfluous newline

File: manifests/tftp.pp [hunk 2]
```diff
@@ -1,32 +1,25 @@
...
-  foreman_proxy::tftp::copy_file{$foreman_proxy::tftp_syslinux_filenames:
+  foreman_proxy::tftp::copy_file { $foreman_proxy::tftp_syslinux_filenames:
     target_path => $foreman_proxy::tftp_root,
-    require     => $required_tftp_classes;
+    require     => Class['foreman_proxy::install', 'tftp::install'],
   }
 
   if $foreman_proxy::tftp_manage_wget {
     ensure_packages(['wget'], { ensure => $foreman_proxy::ensure_packages_version, })
   }
```
Comment (mmoll) [inline manifests/tftp.pp]: superfluous newline

File: manifests/tftp.pp [hunk 3]
```diff
@@ -1,32 +1,25 @@
 # Set up the tftp component
-class foreman_proxy::tftp (
-  $tftp_managed = $foreman_proxy::tftp_managed,
-) {
-  $required_tftp_classes = $tftp_managed ? {
-    true    => Class['foreman_proxy::install', 'tftp::install'],
-    default => Class['foreman_proxy::install'],
+class foreman_proxy::tftp {
+
```
Comment (mmoll) [inline manifests/tftp.pp]: superfluous newline

**Comment by mmoll** [comment]
please update (or best extend) the tests, I cancelled all except the 2.3.0 run of this commit to save resources

**Comment by dc-mattj** [comment]
Tests updated

**Comment by domcleal** [comment]
Thanks @dc-mattj!

**Comment by ekohl** [comment]
I wonder if we should provide a ```tftp_provider```. This could allow using libvirt but initially a noop provider that just places the files could help https://github.com/Katello/forklift/issues/295

**Comment by mmoll** [comment]
As for the proxy side there's nothing really changing, maybe we should have a "libvirt" and a "tftpd" backend in puppet-tftp?

**Comment by stbenjam** [comment]
I don't really know if a libvirt provider is useful to anyone in puppet-tftp, as the only useful thing being done for libvirt is creation and population of the directories and such which is done here.

**Comment by mmoll** [comment]
the steps of editing the libvirtd XML could be automated so tftp gets enabled

**Pull request merged**